src/databricks/labs/dqx/profiler/generator.py

```diff
@@ -45,6 +81,54 @@ def generate_dq_rules(self, profiles: list[DQProfile] | None = None, level: str
 
         return dq_rules
 
+    @telemetry_logger("generator", "generate_dq_rules_with_llm")
+    def generate_dq_rules_with_llm(self, user_input: str, table_name: str = "") -> list[dict]:
+        """
+        Generates data quality rules using LLM based on natural language input.
+
+        Args:
+            user_input: Natural language description of data quality requirements.
+            table_name: Optional fully qualified table name.
+                        If not provided, LLM will be used to guess the table schema.
+
+        Returns:
+            A list of dictionaries representing the generated data quality rules.
+
+        Raises:
+            RuntimeError: If DSPy compiler is not available.
+        """
+        # Check if DSPy compiler is available
+        if self.dspy_compiler is None:
+            raise RuntimeError(
```

Comment:
```suggestion
            raise MissingParameterError(
```
from databricks.labs.dqx.errors import MissingParameterError

let's use custom exception  here